httpserver/handlers.go

```diff
@@ -62,20 +62,21 @@ func (server *Server) verify(ctx context.Context, w http.ResponseWriter, r *http
 	mu := sync.Mutex{}
```

Comment:
@binbin-li since you are in this function, can you add some comments at the top of the Verify function that indicates what this function verifies